support/cas-server-support-syncope-authentication/src/main/java/org/apereo/cas/syncope/SyncopeAuthenticationHandler.java

```diff
@@ -62,42 +64,45 @@ protected AuthenticationHandlerExecutionResult authenticateUsernamePasswordInter
         val result = authenticateSyncopeUser(credential);
         if (result.isPresent()) {
-            val user = result.get();
+            JsonNode user = result.get();
             LOGGER.debug("Received user object as [{}]", user);
-            if (user.has("suspended") && user.get("suspended").asBoolean()) {
-                throw new AccountDisabledException(
-                        "Could not authenticate forbidden account for " + credential.getUsername());
-            }
-            if (user.has("mustChangePassword") && user.get("mustChangePassword").asBoolean()) {
-                throw new AccountPasswordMustChangeException(
-                        "Account password must change for " + credential.getUsername());
-            }
-            val principal = principalFactory.createPrincipal(user.get("username").asText(),
+            Principal principal = principalFactory.createPrincipal(user.get("username").asText(),
                     SyncopeUtils.convertFromUserEntity(user, properties.getAttributeMappings()));
             return createHandlerResult(credential, principal, new ArrayList<>());
         }
         throw new FailedLoginException("Could not authenticate account for " + credential.getUsername());
     }
 
-    protected Optional<JsonNode> authenticateSyncopeUser(final UsernamePasswordCredential credential) {
+    protected Optional<JsonNode> authenticateSyncopeUser(final UsernamePasswordCredential credential)
+            throws AccountPasswordMustChangeException, AccountDisabledException {
         HttpResponse response = null;
         try {
-            val syncopeRestUrl = StringUtils.appendIfMissing(
+            String syncopeRestUrl = StringUtils.appendIfMissing(
                     SpringExpressionLanguageValueResolver.getInstance().resolve(properties.getUrl()),
                     "/rest/users/self");
-            val exec = HttpExecutionRequest.builder()
+            HttpExecutionRequest exec = HttpExecutionRequest.builder()
                     .method(HttpMethod.GET)
                     .url(syncopeRestUrl)
                     .basicAuthUsername(credential.getUsername())
                     .basicAuthPassword(credential.toPassword())
                     .headers(CollectionUtils.wrap("X-Syncope-Domain", syncopeDomain))
                     .maximumRetryAttempts(properties.getMaxRetryAttempts())
                     .build();
-            response = Objects.requireNonNull(HttpUtils.execute(exec));
+            response = Objects.requireNonNull(SyncopeUtils.execute(exec));
             LOGGER.debug("Received http response status as [{}]", response.getReasonPhrase());
             if (response.getCode() == HttpStatus.SC_OK) {
                 return parseResponseResults((HttpEntityContainer) response);
             }
+            val header = response.getHeader("x-application-error-info");
+            if (header.getValue().contains("change your password")) {
```

Comment:
This will do the search in case-sensitive mode. I think you need to adjust to also handle case insensitivity, just in case.